src/main/scala/io/github/pauljamescleary/petstore/domain/users/UserService.scala

```diff
@@ -14,7 +14,7 @@ class UserService[F[_]: Monad](userRepo: UserRepositoryAlgebra[F], validation: U
     } yield saved
 
   def getUser(userId: Long): EitherT[F, UserNotFoundError.type, User] =
-    EitherT.fromOptionF(userRepo.get(userId), UserNotFoundError)
+    EitherT.fromOptionF(userRepo.get(userId).value, UserNotFoundError)
```

Comment:
`OptionT` transforms to `EitherT` via `toLeft` and `toRight` on `OptionT`.

Reply:
Appreciate you working with what you had here, and I wrote this `EitherT[F, UserNotFoundError.type, User]` thing last year apparently. I'm trying to figure out if this API makes sense now. Why have `EitherT[F, L, R]` where L is fixed to a `case object` type? This `L` provides no extra information compared to `OptionT.none`. Perhaps we should just use this opporunity to simplify this api too. Let me know what you think.

Reply:
For this use case, having EitherT doesn't add value in the short time. But, I'd rather not change it as if we want to add new errors on the future we will have to change all instances of OptionT to EitherT. Also, replacing EitherT for OptionT won't simplify  the api much..